Islandora/islandora_scholar
# updates for php 7.2

**JIRA Ticket**: https://jira.duraspace.org/browse/ISLANDORA-2414

# What does this Pull Request do?

A couple of updates for PHP 7.2+ compatibility

# What's new?

Smaller updates:

* A few of the functions using `each()` in `modules/citeproc/includes/convert.inc` have been updated to use PHP 7.2+ friendly functionality.
* `islandora_bibliography`'s untracked dependency on `citation_exporter` is now tracked

Larger update:

* TCPDF has been removed from this repo and re-implemented as a proper library. The existing version of TCPDF bundled with `citation_exporter` is now 7 years old and no longer PHP 7.2+ compatible. More recent versions of the repo claim support for PHP 7.3. Functionality appears to be the same with version 6.2.25

# How should this be tested?

Setup:
* Use PHP 7.2+ for testing
* Clone https://github.com/tecnickcom/TCPDF to your site's libraries folder.

Upon ingesting one or more citation objects:
* When using the citeproc module to render metadata, no difference should be encountered with metadata display before and after the update
* After creating a bookmark containing these citation objects, no difference should be encountered when exporting bookmarks as a PDF before and after the update.

# Additional Notes:

This update does require people to clone down the TCPDF library, so there is some tiny update overhead involved with this beyond just the standard pull/clear cache. This is pretty standard stuff though

# Interested parties
@DonRichards and @bryjbrown as the maintainers of the respective modules, also @Islandora/7-x-1-x-committers in general as this would require some intervention from systems folk upon updating
